Scripts/RemediationScripts/Remediate-ConfigureIoTHubTLSVersion.ps1

```diff
@@ -0,0 +1,878 @@
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Minimum required TLS Version configured set on IoT Hub.")    
+            $logSkippedResources += $logResource
+        }
+    }
+
+    $totalNonCompliantIoTHub = ($NonCompliantTLSIoTHub | Measure-Object).Count
+
+    if ($totalNonCompliantIoTHub -eq 0)
+    {
+        Write-Host "No IoT Hub(s) found with non-compliant TLS Version less than [$($requiredMinTLSVersion)]. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+
+    Write-Host "Found [$($totalNonCompliantIoTHub)] IoT Hub(s) with non-compliant TLS Version:" -ForegroundColor $([Constants]::MessageType.Update)
+
+    $colsProperty = @{Expression={$_.ResourceName};Label="ResourceName";Width=30;Alignment="left"},
+                    @{Expression={$_.ResourceGroupName};Label="ResourceGroupName";Width=30;Alignment="left"},
+                    @{Expression={$_.ResourceId};Label="ResourceId";Width=50;Alignment="left"},
+                    @{Expression={$_.MinimumTlsVersion};Label="MinimumTLSVersion";Width=10;Alignment="left"}
+        
+    $NonCompliantTLSIoTHub | Format-Table -Property $colsProperty -Wrap
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\MinTLSVersionForIoTHub"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up IoT Hub(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        # Backing up IoT Hub(s) details
+        $backupFile = "$($backupFolderPath)\NonCompliantTLSIoTHub.csv"
+
+        $NonCompliantTLSIoTHub | Export-CSV -Path $backupFile -NoTypeInformation
+
+        Write-Host "IoT Hub(s) details have been backed up to" -NoNewline
+        Write-Host " [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else
+    {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+    }
+
+    if (-not $DryRun)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Remediating non-compliant Azure IoT Hubs..." 
+        Write-Host $([Constants]::SingleDashLine)
+        
+        if(-not $AutoRemediation)
+        {
+            if (-not $Force)
+            {
+                Write-Host "Found total [$($TLSwithDefaultValue.count)] IoT Hub(s) with default TLS version. TLS version for these resources cannot be reverted back to default value after remediation." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
cross check on this message

Reply:
Removed that message as it is not needed/applicable on this BRS